hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1480,25 +1480,27 @@ private class Emit(
         val requiredData = dataPType.checkedConvertFrom(mb, region, datat.value[Long], dataContainer, "NDArray cannot have missing data")
         val shapeSrvb = new StagedRegionValueBuilder(mb, targetShapePType)
         val ndAddress = mb.newField[Long]
+        val shapeAddress = mb.newField[Long]
 
 
-        def getShapeAtIdx(index: Int) = Region.loadLong(shapePType.loadField(shapet.value[Long], index))
+        def getShapeAtIdx(index: Int) = Region.loadLong(shapePType.loadField(shapeAddress, index))
 
         val setup = Code(
           shapet.setup,
           datat.setup,
           rowMajort.setup,
+          shapeAddress := shapet.value[Long],
```

Comment:
you can't run the value before you check for missingness, push this into the false branch of the subsequent mux.